app/assets/javascripts/discourse/app/mixins/uppy-upload.js

```diff
@@ -396,9 +396,9 @@ export default Mixin.create(UppyS3Multipart, ExtendableUploader, {
             return {
               method: "put",
               url: response.url,
-              headers: {
+              headers: deepMerge(response.signed_headers, {
```

Comment:
Are `signed_headers` a deeply nested object? If not then this can be:

```suggestion
              headers: {
                ...response.signed_headers,
```

and later:
```diff
- }),
+ },
```

Reply:
Ah yes much better, thanks!